shared/actions/devices.js

```diff
@@ -1,138 +1,192 @@
 // @flow
 import * as Constants from '../constants/devices'
 import HiddenString from '../util/hidden-string'
-import type {AsyncAction} from '../constants/types/flux'
 import {Map} from 'immutable'
 import {devicesTab, loginTab} from '../constants/tabs'
-import {loginDeprovisionRpc, revokeRevokeDeviceRpc, deviceDeviceHistoryListRpc,
-  loginPaperKeyRpc} from '../constants/types/flow-types'
+
 import {navigateTo, navigateUp, switchTab} from './router'
 import {setRevokedSelf} from './login'
+import {buffers, eventChannel, END, takeEvery, takeLatest} from 'redux-saga'
+import {call, put, select, take} from 'redux-saga/effects'
+import {
+  deviceDeviceHistoryListRpcPromise,
+  loginDeprovisionRpcPromise,
+  loginPaperKeyRpc,
+  revokeRevokeDeviceRpcPromise,
+} from '../constants/types/flow-types'
 
-export function loadDevices () : AsyncAction {
-  return (dispatch) => {
-    dispatch({
-      type: Constants.loadingDevices,
-      payload: null,
-    })
+import type {
+  DeviceRemoved,
+  GeneratePaperKey,
+  IncomingDisplayPaperKeyPhrase,
+  LoadDevices,
+  LoadingDevices,
+  PaperKeyLoaded,
+  PaperKeyLoading,
+  RemoveDevice,
+  ShowDevices,
+} from '../constants/devices'
+import type {SagaGenerator} from '../constants/types/saga'
 
-    deviceDeviceHistoryListRpc({
-      callback: (error, devices) => {
-        // Flow is weird here, we have to give it true or false directly instead of just giving it !!error
-        if (error) {
-          dispatch({
-            type: Constants.showDevices,
-            payload: error,
-            error: true,
-          })
-        } else {
-          dispatch({
-            type: Constants.showDevices,
-            payload: devices,
-            error: false,
-          })
-        }
-      },
-    })
-  }
+export function loadDevices (): LoadDevices {
+  return {type: Constants.loadDevices, payload: undefined}
 }
 
-export function generatePaperKey () : AsyncAction {
-  return (dispatch) => {
-    dispatch({
-      type: Constants.paperKeyLoading,
-      payload: null,
-    })
+export function loadingDevices (): LoadingDevices {
```

Comment:
ah weird, I feel like it should have a `{waiting: boolean}` type for payload or something. This is fine for now